packages/components/src/TreeView/TreeViewItem/TreeViewItem.component.js

```diff
@@ -34,122 +37,139 @@ function getActionHandler(func, item) {
  * @returns XML, jsx to display
  */
 
-function TreeViewItem({ id, item, depth = 0, onClick, onSelect }) {
-	const {
-		toggled = false,
-		selected,
-		hidden,
-		name,
-		children = [],
-		showCounter,
-		actions,
-		icon = 'talend-folder',
-		counter = children.length,
-	} = item;
-	const toggleIconLabel = toggled ? 'Collapse' : 'Expand';
-
-	function onSelectWrapper() {
-		return onSelect(item);
+class TreeViewItem extends React.Component {
+	static propTypes = {
+		id: PropTypes.string.isRequired,
+		item: PropTypes.shape({
+			name: PropTypes.string.isRequired,
+			toggled: PropTypes.bool,
+			selected: PropTypes.bool,
+			children: PropTypes.arrayOf(PropTypes.object),
+			icon: PropTypes.string,
+			actions: PropTypes.arrayOf(
+				PropTypes.shape({
+					action: PropTypes.func,
+					label: PropTypes.string,
+					icon: PropTypes.string,
+				}),
+			),
+			counter: PropTypes.number,
+			showCounter: PropTypes.bool,
+		}).isRequired,
+		onClick: PropTypes.func.isRequired,
+		onSelect: PropTypes.func.isRequired,
+		depth: PropTypes.number,
+	};
+
+	static defaultProps = {
+		id: 'tc-treeview-item',
+		depth: 0,
+	};
+
+	constructor(props) {
+		super(props);
+		this.onSelect = this.onSelect.bind(this);
+		this.onKeyDown = this.onKeyDown.bind(this);
+		this.renderTreeViewItem = this.renderTreeViewItem.bind(this);
 	}
 
-	function getTreeViewItem(child, i) {
+	onKeyDown(event) {
+		if (event.keyCode === keycode.codes.enter) {
```

Comment:
Shouldn't we support `space` key too ? (Just an idea)

Reply:
done in #880